code-orange/Communicator.swift

```diff
@@ -64,3 +75,25 @@ extension Communicator: DataFetcher {
     return infectedLocations
   }
 }
+
+extension Communicator: DataSetter {
+  func shareInfectedLocations(_ locations: [COLocation]) {
+    guard let url = URL(string: serverUrl) else { return }
+    var json = [String: Codable]()
+    json["patientStatus"] = "carrier"
+    json["country"] = "il"
+    json["locations"] = locations
+    
+    guard let data = try? jsonEncoder.encode(locations) else {
+      print("Error encoding locations")
```

Comment:
Consider catching the error and print a message with the error for debugging purposes

Reply:
done!